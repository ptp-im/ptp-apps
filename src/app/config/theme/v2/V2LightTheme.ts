import color from 'color';
import { MD2LightTheme } from 'react-native-paper';
import type { MD2Theme } from 'react-native-paper';

import { black, red800, white } from './colors';

export const V2LightTheme: MD2Theme = {
  ...MD2LightTheme,
  dark: false,
  roundness: 4,
  version: 2,
  isV3: false,
  colors: {
    ...MD2LightTheme.colors,
    // primary: '#6200ee',
    accent: '#03dac4',
    // background: '#f6f6f6',
    // background: grey50,
    surface: white,
    error: '#B00020',
    // text: black,
    // onSurface: '#000000',
    disabled: color(black).alpha(0.26).rgb().string(),
    placeholder: color(black).alpha(0.54).rgb().string(),
    backdrop: color(black).alpha(0.5).rgb().string(),
    // notification: pinkA400,

    msgUnreadBadge: red800,
    onSurface: 'rgb(30, 107, 192)',
    primary: 'rgb(30, 107, 192)',
    background: 'rgb(235, 235, 240)',
    white: 'rgb(255, 255, 255)',
    background1: 'rgb(235, 235, 240)',
    backgroundSelected: 'rgb(62, 125, 185)',
    backgroundTabs: 'rgb(255, 255, 255)',
    card: 'rgb(255, 255, 255)',
    text: 'rgb(28, 28, 30)',
    textPrimary: 'rgb(10, 132, 255)',
    textAlert: 'rgb(255, 59, 48)',
    border: 'rgb(229, 229, 229)',
    notification: 'rgb(255, 59, 48)',
  },
};

export const V2NavLightColors = {
  primary: V2LightTheme.colors.primary,
  background: V2LightTheme.colors.background,
  card: V2LightTheme.colors.card,
  text: V2LightTheme.colors.text,
  border: V2LightTheme.colors.border,
  notification: V2LightTheme.colors.notification,
};
