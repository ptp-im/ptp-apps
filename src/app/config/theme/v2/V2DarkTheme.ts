import color from 'color';

import type { MD2Theme } from 'react-native-paper';
import { black, red800, white } from './colors';
import { V2LightTheme } from './V2LightTheme';

export const V2DarkTheme: MD2Theme = {
  ...V2LightTheme,
  dark: true,
  mode: 'adaptive',
  version: 2,
  isV3: false,
  colors: {
    ...V2LightTheme.colors,
    // primary: '#BB86FC',
    accent: '#03dac6',
    // background: '#121212',
    surface: '#121212',
    error: '#CF6679',
    onSurface: '#FFFFFF',
    // text: white,
    disabled: color(white).alpha(0.38).rgb().string(),
    placeholder: color(white).alpha(0.54).rgb().string(),
    backdrop: color(black).alpha(0.5).rgb().string(),
    // notification: pinkA100,

    msgUnreadBadge: red800,
    primary: 'rgb(10, 132, 255)',
    background: 'rgb(1, 1, 1)',
    card: 'rgb(18, 18, 18)',
    text: 'rgb(229, 229, 231)',
    textPrimary: 'rgb(10, 132, 255)',
    border: 'rgb(39, 39, 41)',
    notification: 'rgb(255, 69, 58)',
  },
};

export const V2NavDarkColors = {
  primary: V2LightTheme.colors.primary,
  background: V2LightTheme.colors.background,
  card: V2LightTheme.colors.card,
  text: V2LightTheme.colors.text,
  border: V2LightTheme.colors.border,
  notification: V2LightTheme.colors.notification,
};
