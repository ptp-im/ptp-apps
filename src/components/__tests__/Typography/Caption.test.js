import * as React from 'react';
import { StyleSheet } from 'react-native';

import renderer from 'react-test-renderer';

import { red500 } from '../../../styles/themes/v2/colors';
import Caption from '../../Typography/v2/Caption.tsx';

const style = StyleSheet.create({
  caption: {
    fontSize: 20,
    color: red500,
  },
});

it('renders caption with children as content', () => {
  const tree = renderer.create(<Caption>Caption content</Caption>).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders caption applying style', () => {
  const tree = renderer
    .create(<Caption style={style.caption}>Big and red caption</Caption>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
