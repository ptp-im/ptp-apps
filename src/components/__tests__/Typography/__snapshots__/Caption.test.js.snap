// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders caption applying style 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "rgba(28, 27, 31, 1)",
      },
      Object {
        "textAlign": "left",
      },
      Array [
        Object {
          "textAlign": "left",
        },
        Object {
          "color": "rgba(28, 27, 31, 0.54)",
          "writingDirection": "ltr",
        },
        Array [
          Object {
            "fontSize": 12,
            "letterSpacing": 0.4,
            "lineHeight": 20,
            "marginVertical": 2,
          },
          Object {
            "color": "#f44336",
            "fontSize": 20,
          },
        ],
      ],
    ]
  }
>
  Big and red caption
</Text>
`;

exports[`renders caption with children as content 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "rgba(28, 27, 31, 1)",
      },
      Object {
        "textAlign": "left",
      },
      Array [
        Object {
          "textAlign": "left",
        },
        Object {
          "color": "rgba(28, 27, 31, 0.54)",
          "writingDirection": "ltr",
        },
        Array [
          Object {
            "fontSize": 12,
            "letterSpacing": 0.4,
            "lineHeight": 20,
            "marginVertical": 2,
          },
          undefined,
        ],
      ],
    ]
  }
>
  Caption content
</Text>
`;
